### PR TITLE
fix: make `${file(.ts)}` and `serverless diff` work in the published distribution

### PR DESCRIPTION
**Pull request by czubocha**
## Summary

Two features from 4.37.0 work when running the framework from source but fail against the published distribution because of how the bundled code locates external files at runtime. This PR addresses both.

### TypeScript files in `${file(...)}`

Referencing a TypeScript module from a configuration variable failed with:

```
✖ Cannot load TS module "./mod.ts":
  ERR_MODULE_NOT_FOUND ...
```

Switch the TypeScript-file loader to use the same path the framework already uses successfully for `serverless.ts`. Behavior, supported export shapes, and module-shape interop are unchanged.

### `serverless diff`

Running the new `serverless diff` command failed with:

```
✖ ENOENT: no such file or directory, open '<package>/db.json.gz'
```

The command depends on a data asset that needs to ship alongside the bundled code. Copy it to the location the bundled code looks up at runtime.

## Before / after

| Scenario | Before | After |
|---|---|---|
| `${file(./mod.ts):export}` referenced from `serverless.yml` | crashes with `ERR_MODULE_NOT_FOUND` | resolves the export value as documented |
| `serverless diff` against a deployed stack | crashes with `ENOENT: ... db.json.gz` | renders the structured diff as documented |

## Test plan

- [x] TypeScript file-resolver unit tests pass (18/18 in `file-ts.test.js`).
- [x] Full file-resolver test suite passes (328/328 in `tests/unit/resolvers/`).
- [x] `npm run prettier` clean.
- [x] `npm run lint` clean.
- [x] End-to-end against a real service: `${file(./secrets.ts):getSecrets}` chained through `${self:custom.secrets.apiKey}` resolves correctly and propagates to the function's environment variable.
- [x] All three TypeScript export shapes verified through the existing `__esModule` unwrap test suite (default-only, named-only, mixed).
- [ ] After this PR is built into a new canary, rerun `serverless diff` from the canary against a deployed stack and confirm structured output.

## Summary by CodeRabbit

* **Refactor**
  * Switched TypeScript module loading to a CommonJS runtime API for more reliable TS module execution.

* **Tests**
  * Updated unit tests for the TS resolver to match the new loading behavior and unwrap/edge-case scenarios.

* **Chores**
  * Packaging and distribution now include the resource-spec database so serverless diff can locate it at runtime.

[Review Change Stack](https://app.coderabbit.ai/change-stack/serverless/serverless/pull/13609?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by Mmarzex** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.us.snyk.io/org/framework/pr-checks/ee0307d1-a2e1-4dcf-a277-cfc8748ec085?refs=7eb0cd767ce707291558f3acbe39d3c111ec32ec&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.us.snyk.io/org/framework/pr-checks/ee0307d1-a2e1-4dcf-a277-cfc8748ec085/license?refs=7eb0cd767ce707291558f3acbe39d3c111ec32ec&source=prChecksComment) |
| :white_check_mark: | Code Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.us.snyk.io/org/framework/project/c9cc4ca0-1caa-42d1-9aa9-df2e9b76ea90/pr-check/bd639d94-a0ca-48f6-918b-7ea69b57f393?refs=7eb0cd767ce707291558f3acbe39d3c111ec32ec&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/sf-core/scripts/prepareDistributionTarballs.js [hunk 1]
```diff
@@ -97,6 +97,19 @@ cpPromises.push(
...
+// from inside the bundled code; when sf-core.js is loaded from
+// `<package>/dist/`, that path resolves to `<package>/db.json.gz`. Copy it
+// there so the diff command can render structured output against deployed
+// stacks.
+cpPromises.push(
+  cp(
+    resolveFromServerless.resolve('@aws-cdk/aws-service-spec/db.json.gz'),
+    '../../framework-dist/db.json.gz',
+  ),
+)
```
[the remark this answers is not in the document]
Comment (czubocha) [reply packages/sf-core/scripts/prepareDistributionTarballs.js:111]: Verified empirically - resolveFromServerless.resolve('@aws-cdk/aws-service-spec/db.json.gz') succeeds and returns the expected path. The upstream package has no exports field, so legacy subpath resolution applies and any file in the package dir is reachable. The full npm run test:build run also exercises this code path and produces db.json.gz in the tarball at the expected byte size.

On the transitive-dep observation: package-lock.json already pins the resolution by integrity hash, and if the parent ever restructures, resolve() throws synchronously with a clear Cannot find module '@aws-cdk/aws-service-spec/db.json.gz' from '…' — which halts the build loudly. Declaring it directly would just introduce a version-pin we'd need to keep in sync with whatever @aws-cdk/cloudformation-diff ships. Leaving as-is.
